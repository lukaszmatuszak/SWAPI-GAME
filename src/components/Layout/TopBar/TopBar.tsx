import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const TopBar: React.FC = () => (
    <AppBar position="static">
        <Toolbar variant="dense">
            <Typography variant="h6" color="inherit">
                SWAPI Game
            </Typography>
        </Toolbar>
    </AppBar>
);

export default TopBar;
