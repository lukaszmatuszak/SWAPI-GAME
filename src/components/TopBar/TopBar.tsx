import React from 'react';
import { AppBar, Box, Toolbar, Typography } from '@material-ui/core';

import ChangeLanguageMenu from '../ChangeLanguageMenu/ChangeLanguageMenu';

const TopBar: React.FC = () => (
    <AppBar position="static">
        <Toolbar variant="dense">
            <Box flexGrow={1}>
                <Typography variant="h6" color="inherit">
                    SWAPI Game
                </Typography>
            </Box>
            <ChangeLanguageMenu />
        </Toolbar>
    </AppBar>
);

export default TopBar;
