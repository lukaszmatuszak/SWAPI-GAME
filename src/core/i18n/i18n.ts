import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import locales from './locales/index';

i18n.use(initReactI18next)
    .use(LanguageDetector)
    .init({
        ns: ['Common'],
        defaultNS: 'Common',
        resources: locales,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false,
        },
    });
