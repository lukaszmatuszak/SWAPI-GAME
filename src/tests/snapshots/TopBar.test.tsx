import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TopBar from '../../components/TopBar/TopBar';
import '../../core/i18n/i18n';

describe('TopBar Snapshot - ', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<TopBar />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
