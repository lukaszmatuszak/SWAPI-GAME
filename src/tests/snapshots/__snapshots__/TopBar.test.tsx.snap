// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Snapshot -  renders correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))
    variant="dense"
  >
    <Styled(MuiBox)
      flexGrow={1}
    >
      <Link
        to="/"
      >
        <WithStyles(ForwardRef(Typography))
          color="inherit"
          variant="h6"
        >
          SWAPI Game
        </WithStyles(ForwardRef(Typography))>
      </Link>
    </Styled(MuiBox)>
    <ChangeLanguageMenu />
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
